ts/features/bonus/bpd/components/superCashbackRanking/SuperCashbackRanking.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import { View } from "native-base";
+import { fromNullable } from "fp-ts/lib/Option";
+import { GlobalState } from "../../../../../store/reducers/types";
+import { bpdSelectedPeriodSelector } from "../../store/reducers/details/selectedPeriod";
+import ItemSeparatorComponent from "../../../../../components/ItemSeparatorComponent";
+import { H3 } from "../../../../../components/core/typography/H3";
+import I18n from "../../../../../i18n";
+import Markdown from "../../../../../components/ui/Markdown";
+import { IOColors } from "../../../../../components/core/variables/IOColors";
+import { BpdPeriodWithInfo } from "../../store/reducers/details/periods";
+import { localeDateFormat } from "../../../../../utils/locale";
+import { FirstPositionItem } from "./FirstPositionItem";
+import { LastPositionItem } from "./LastPositionItem";
+import UserPositionItem from "./UserPositionItem";
+
+type Props = ReturnType<typeof mapStateToProps>;
+
+type tmpCitizenRankingResource = {
+  totalParticipants: number;
+  ranking: number;
+  // Numero massimo di transazioni effettuate dagli utenti che rientrano nel 'rimborso speciale'
+  maxTransactionNumber: number;
+  // Numero minimo di transazioni effettuate dagli utenti che rientrano nel 'rimborso speciale'
+  minTransactionNumber: number;
+  transactionNumber: number;
+  awardPeriodId: number;
+};
+
+const RankingItems: React.FunctionComponent<Props> = (props: Props) => {
+  if (props.selectedPeriod) {
+    if (props.selectedPeriod.minPosition < props.rankingInformation.ranking) {
+      return (
+        <>
+          <FirstPositionItem
```

Comment:
Maybe we could avoid all this code where it is possible.
For example FirstPositionItem seems to be always the same, so we could do


```typescript
const firstPositionItem = (
      <FirstPositionItem
        superCashbackAmount={props.selectedPeriod.superCashbackAmount}
        transactionsNumber={props.rankingInformation.maxTransactionNumber}
      />
    );
```

```suggestion
          {firstPositionItem}
```

Reply:
@Undermaken I made some refactoring to this function in 8ec14bd can you check this again